src/Stan/Hie/MatchAst.hs

```diff
@@ -63,6 +71,12 @@ hieMatchPatternAst hie@HieFile{..} node@Node{..} = \case
     PatternAstNodeExact tags patChildren ->
            matchAnnotations tags nodeInfo
         && checkWith (hieMatchPatternAst hie) nodeChildren patChildren
+    PatternAstVarName varName -> isJust $ find
+        (\(k, _) -> case k of
+            Right x -> varName `Str.isInfixOf` map toLower (occNameString $ nameOccName x)
+            _       -> False
+        )
+        $ Map.toList $ nodeIdentifiers nodeInfo
```

Comment:
I guess, if you do `Map.toKeys`, you can use `LambdaCase` in the predicate above